modules/tinymce/src/themes/silver/test/ts/browser/editor/color/TextColorSanityTest.ts

```diff
@@ -88,4 +90,27 @@ describe('browser.tinymce.themes.silver.editor.color.TextColorSanityTest', () =>
     TinyUiActions.clickOnUi(editor, 'div[data-mce-color="#236FA1"]');
     TinyAssertions.assertContentStructure(editor, backcolorStruct('rgb(35, 111, 161)'));
   });
+
+  it('TINY-9184: Adding a color is added as expected', async () => {
+    const editor = hook.editor();
+    setupContent(editor);
+    TinyUiActions.clickOnToolbar(editor, '[aria-label="Background color"] > .tox-tbtn + .tox-split-button__chevron');
+    await TinyUiActions.pWaitForUi(editor, '.tox-swatches');
+    UiFinder.notExists(SugarBody.body(), 'div[data-mce-color="#FF0000"]');
+    TinyUiActions.clickOnUi(editor, 'button[title="Custom color"]');
+    const dialog = await TinyUiActions.pWaitForDialog(editor);
+    const input = UiFinder.findIn<HTMLInputElement>(dialog, 'label:contains("R") + input').getOrDie();
+    UiControls.setValue(input, '255');
+    const evt = new Event('input', {
+      bubbles: true,
+      cancelable: true
+    });
+    input.dom.dispatchEvent(evt);
+    await Waiter.pWait(500);
```

Comment:
Why is the wait needed here is the input some how async? Could you instead await a change of color on the preview or the hex representation to reduce the possibility of a flake here?

Reply:
I don't know exactly why it's needed, but it made things work. I wasn't too concerned about exactly why since it's honestly unrelated to this issue, and I was concerned that trying to wait for the change with some sort of waiter might take some time to implement, since it's in a dialog which requires extra fiddling around, at least for me.

Reply:
Please switch to a waiter if we can. Since if we don't know why it's needed then we might need to know why also if we start sprinkling 500ms waits all over the tests then they start going slow when they start adding up. They already take tons of time to finish we shouldn't add more time to it just because we don't know why something needs to be async. If it's to hard to wait for a state then fine we could use a wait call here but then a comment should be added to why it needs to be there or a follow up ticket reference or something if it's really unknown why it needs to be there there are a few such cases in other parts of the codebase.

Reply:
- [ ] Combine the tests using `context` to remove duplication

Reply:
Agreed

- [ ] Investigate why a 500ms wait works around the problem